.github/workflows/uploadBinaries.yml

```diff
@@ -0,0 +1,122 @@
+      - name: Get LLVM Hash
+        id: get-llvm-hash
+        run: echo "::set-output name=hash::$(git rev-parse @:./llvm)"
+
+      - name: Get workflow spec hash
+        id: get-workflow-hash
+        run: echo "::set-output name=hash::$(md5sum $GITHUB_WORKSPACE/.github/workflows/uploadBinaries.yml | awk '{print $1}')"
+
+      # Try to fetch LLVM from the cache.
+      - name: Cache LLVM
+        id: cache-llvm
+        uses: actions/cache@v2
+        with:
+          path: |
+            llvm/build
+          key: ${{ matrix.runner }}-llvm-${{ steps.get-llvm-hash.outputs.hash }}-${{ steps.get-workflow-hash.outputs.hash }}-${{ matrix.compiler.cc }} 
+
+      - name: Setup Ninja Linux
+        if: matrix.os == 'linux'
+        run: sudo apt-get install ninja-build
+
+      - name: Setup Ninja Mac
+        if: matrix.os == 'macos'
+        run: brew install ninja
+
+      # Build LLVM if we didn't hit in the cache. Even though we build it in
+      # the previous job, there is a low chance that it'll have been evicted by
+      # the time we get here.
+      - name: Rebuild and Install LLVM
+        if: steps.cache-llvm.outputs.cache-hit != 'true'
+        run: |
+          mkdir -p llvm/build
+          cd llvm/build
+          cmake -G Ninja ../llvm \
+              ${{ matrix.cmake-args }} \
+              -DCMAKE_BUILD_TYPE=${{ matrix.compiler.mode }} \
+              -DCMAKE_C_COMPILER=${{ matrix.compiler.cc }} \
+              -DCMAKE_CXX_COMPILER=${{ matrix.compiler.cxx }} \
+              -DBUILD_SHARED_LIBS=${{ matrix.compiler.shared }} \
+              -DLLVM_BUILD_EXAMPLES=OFF \
+              -DLLVM_ENABLE_ASSERTIONS=${{ matrix.compiler.assert }} \
+              -DLLVM_ENABLE_BINDINGS=OFF \
+              -DLLVM_ENABLE_OCAMLDOC=OFF \
+              -DLLVM_ENABLE_PROJECTS='mlir' \
+              -DLLVM_OPTIMIZED_TABLEGEN=ON \
+              -DLLVM_STATIC_LINK_CXX_STDLIB=ON \
+              -DLLVM_ENABLE_TERMINFO=OFF \
+              -DLLVM_PARALLEL_LINK_JOBS=1 \
+              -DLLVM_TARGETS_TO_BUILD="host"
+          ninja
+
+      # --------
+      # Build and test CIRCT
+      # --------
+
+      - name: Build and Test CIRCT
+        run: |
+          mkdir build
+          cd build
+          cmake -G Ninja .. \
+            ${{ matrix.cmake-args }} \
+            -DBUILD_SHARED_LIBS=${{ matrix.compiler.shared }} \
+            -DCMAKE_BUILD_TYPE=${{ matrix.compiler.mode }} \
+            -DLLVM_ENABLE_ASSERTIONS=${{ matrix.compiler.assert }} \
+            -DMLIR_DIR=`pwd`/../llvm/build/lib/cmake/mlir \
+            -DLLVM_DIR=`pwd`/../llvm/build/lib/cmake/llvm \
+            -DCMAKE_C_COMPILER=${{ matrix.compiler.cc }} \
+            -DCMAKE_CXX_COMPILER=${{ matrix.compiler.cxx }} \
+            -DVERILATOR_DISABLE=ON \
+            -DLLVM_ENABLE_TERMINFO=OFF \
+            -DLLVM_STATIC_LINK_CXX_STDLIB=ON \
+            -DLLVM_PARALLEL_LINK_JOBS=1 \
+            -DCMAKE_EXPORT_COMPILE_COMMANDS=OFF
```

Comment:
Does LLVM_TARGETS_TO_BUILD matter for the CIRCT compile?

Reply:
I'm not sure -- for that option I was just following the readme instructions, which say to use it for the LLVM build.